misc/rapiddisk-rootdev/install_initrd.sh

```diff
@@ -104,137 +143,369 @@ if [ -z "$uninstall" ]; then
+	# prepare some vars
 
-			echo " - Copying ${cwd}/ubuntu/rapiddisk to ${bootscript_dest}..."
+	etc_dir_hooks="/etc/initramfs-tools/hooks"
+	etc_dir_scripts="/etc/initramfs-tools/scripts/init-premount"
+	usr_dir_hooks="/usr/share/initramfs-tools/hooks"
+	usr_dir_scripts="/usr/share/initramfs-tools/scripts/init-premount"
 
-			if ! cp "${cwd}/ubuntu/rapiddisk" "${bootscript_dest}"; then
-				# if for any reason the first script was copied with no error, we remove it
-				[ -f "${hook_dest}" ] && rm -f "${hook_dest}" 2>/dev/null
-				myerror "Error: could not copy rapiddisk_hook to ${bootscript_dest}. Exiting..."
-			fi
+	# These checks are intended to find the best place to put the scripts under Ubuntu
 
-			sed -i 's/RAMDISKSIZE/'"${ramdisk_size}"'/' "${bootscript_dest}"
-			sed -i 's,BOOTDEVICE,'"${root_device}"',' "${bootscript_dest}"
+	if [ -d "$etc_dir_hooks" ] && [ -d "$etc_dir_scripts" ]; then
+		hook_dest="${etc_dir_hooks}/rapiddisk_hook_${kernel_version}"
+		bootscript_dest="${etc_dir_scripts}/rapiddisk_${kernel_version}"
+	elif [ -d "$usr_dir_hooks" ] && [ -d "$usr_dir_scripts" ]; then
+		hook_dest="${usr_dir_hooks}/rapiddisk_hook_${kernel_version}"
+		bootscript_dest="${usr_dir_scripts}/rapiddisk_${kernel_version}"
+	else
+		myerror "I can't find any suitable place for initramfs scripts."
+	fi
 
-			chmod +x "${bootscript_dest}" 2>/dev/null
+	if [ "$install_mode" = "simple_install" ] ; then
+		# installing on Ubuntu
+
+		# without --force we just perform this more check
+		if [ -z "$force" ] ; then
+			if [ -x "$hook_dest" ] && [ -x "$bootscript_dest" ] ; then
+				myerror "initrd hook and/or boot scripts are already installed. You should use the '--force' option."
+			fi
 		fi
-	fi
 
-	echo " - Updating initramfs for kernel version ${kernel_version}..."
-	echo ""
-	update-initramfs -u -k "$kernel_version"
-	echo ""
-	echo " - Updating grub..."
-	echo ""
-	update-grub
-	echo ""
-	echo " - Done under Ubuntu. A reboot is needed."
+		# now we can perform some parameters'checks which would be senseless before
+		install_options_checks
+		ubuntu_install
+		ubuntu_end
+		exit 0
+
+	elif [ "$install_mode" = "simple_uninstall" ] ; then
+		echo " - Uninstalling for kernel $kernel_version ${hook_dest}..."
+		if [ -x "$hook_dest" ] || [ ! -z "$force" ] ; then
+			rm -f "$hook_dest"
+			ubuntu_end
+			exit 0
+		else
+			myerror "$hook_dest not present.  You should use the '--force' option."
+		fi
+	elif [ "$install_mode" = "global_uninstall" ] ; then
+		echo " - Global uninstalling for all kernel versions.."
+		echo " - Deleting all files and rebuilding all the initrd files.."
+		hook_install_dir=$(dirname $hook_dest)
+		bootscript_install_dir=$(dirname $bootscript_dest)
+		rm -f "${hook_install_dir}"/rapiddisk* "${bootscript_install_dir}"/rapiddisk*
+		kernel_version=all
+		ubuntu_end
+		exit 0	
+	fi
 
 else
-	myerror "Error: operating system not supported. Exiting..."
+	myerror "unknown OS."
 fi
 
 exit 0
+
```

Comment:
We need to figure out what to do with the entire commented out section below. I would prefer not to merge that.

Reply:
Removed.